client/containers/Search/Search.js

```diff
@@ -16,6 +16,30 @@ const propTypes = {
 	searchData: PropTypes.object.isRequired,
 };
 
+const getSearchPath = (query, page, mode) => {
+	const params = new URLSearchParams();
+
+	if (query) {
+		params.append('q', query);
+	}
+
+	if (page) {
+		params.append('page', page + 1);
+	}
+
+	if (mode !== 'pubs') {
+		params.append('mode', mode);
+	}
```

Comment:
Not the prettiest code, but this seemed like the best way to avoid empty query parameters in the case of no query, e.g. `/search?q=`, or worse, stringified `undefined`s leaking into the URL when using the constructor dictionary initializer.